qiskit_nature/second_q/hamiltonians/lattice_model.py

```diff
@@ -0,0 +1,42 @@
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2021, 2022.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+"""The Lattice Model class."""
+
+import numpy as np
+
+from .hamiltonian import Hamiltonian
+from .lattices import Lattice
+
+
+class LatticeModel(Hamiltonian):
+    """Lattice Model"""
+
+    def __init__(self, lattice: Lattice) -> None:
+        """
+        Args:
+            lattice: Lattice on which the model is defined.
+        """
+        self._lattice = lattice
+
+    @property
+    def lattice(self) -> Lattice:
+        """Return a copy of the input lattice."""
+        return self._lattice.copy()
```

Comment:
Is there a reason to return a copy (its actually an object deepcopy looking at the implementation), rather than by reference which would allow the lattice to be updated later in the model? I see this is what was done before though.

Reply:
I have not given that much thought tbh. I guess providing write-access would allow flexibility although the user would need to be careful with the changes.

Do you think we should change it to not copy the lattice?